### PR TITLE
deps: bump emotion in recipe and doc site

### PR DESCRIPTION
**Pull request by noranda**
## What did we change?

Upgraded the following packages:

- `@emotion/styled` from `11.6.0` to `11.8.1` in recipe and from `10.0.11` to `11.8.1` in the doc site
- `@emotion/react` from `11.7.1` to `11.8.1` in recipe and from `10.0.10` to `11.8.1` in the doc site - Note: [version 11](https://emotion.sh/docs/emotion-11) renamed the package from `@emotion/core` to `@emotion/react`
- `@emotion/is-prop-valid` from `0.7.3` to `1.1.2` in recipe, not used in doc site

As part of the upgrade, also refactored `EzButton` not to pass through `loading` prop to the underlying HTML button element. `isPropValid` considers `loading` to be a valid HTML attribute, and while it is a valid HTML attribute for some tags, it is not a valid HTML attribute for the button tag.

Edit: Removed `emotion-theming` as `ThemeProvider` can now be imported from `@emotion/react`.

## Why are we doing this?

In order for MUI themes to work correctly in the doc site, emotion needed to be bumped to version 11 to match MUI's emotion version. Bumped all emotion dependencies for consistency.

## Checklist

- [x] Create a changeset (`npm run changeset`) with a summary of the changes

### CODE REVIEW

File: packages/snapshot/snapshot.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const fs = require('fs');
 const React = require('react');
 const Components = require('@ezcater/recipe');
 const styled = require('@emotion/styled').default;
-const {Global, css} = require('@emotion/core');
+const {Global, css} = require('@emotion/react');
 const {ThemeProvider} = require('emotion-theming');
```
Comment (pgrippi) [inline packages/snapshot/snapshot.js:7]: `ThemeProvider` is also provided by `@emotion/react` (I think this was part of the 10 -> 11 emotion upgrade), I think you can move this provider up into the live above and remove `emotion-theming` from the package.json altogether

**Review by pgrippi — APPROVED** [review]

File: packages/snapshot/snapshot.js (hunk 1 above)
Comment (noranda) [reply packages/snapshot/snapshot.js]: Great suggestion, thanks!

**Review by pgrippi — APPROVED** [review]

**Review by pgrippi — DISMISSED** [review]

**Review by pgrippi — APPROVED** [review]

**Review by danidewitt — COMMENTED** [review]
aside from my comments below about changesets, everything else looks good.

1. for the deps changeset, could this be [breaking change](https://github.com/ezcater/recipe/blob/main/doc-site/src/pages/guides/versioning-and-changelog.md) in minimum dependencies? 
2. should we add a `refactor` changeset for `EzButton`?

**Review by danidewitt — DISMISSED** [review]

File: src/components/EzButton/EzButton.tsx [hunk 2]
```diff
@@ -264,12 +264,12 @@ const disabledButtonWrapper = theme.css({display: 'inline-block'});
  * Buttons represent actions on a page that can be triggered with one click.
  * Buttons can be used in forms, or in other locations in a page to communicate that an action is available.
  */
-const EzButton = forwardRef<Ref, Props>((initProps: any, ref) => {
-  const props = {...initProps, disabled: initProps.disabled || initProps.loading};
+const EzButton = forwardRef<Ref, Props>(({loading, ...initProps}: any, ref) => {
```
Comment (danidewitt) [inline src/components/EzButton/EzButton.tsx:267]: should we add a `refactor` changeset for EzButton?

File: package.json [hunk 3]
```diff
@@ -114,10 +114,10 @@
...
   },
   "dependencies": {
-    "@emotion/is-prop-valid": "^0.7.3",
-    "@emotion/react": "^11.7.1",
-    "@emotion/styled": "^11.6.0",
-    "@ezcater/snitches": "^0.2.9",
+    "@emotion/is-prop-valid": "^1.1.2",
+    "@emotion/react": "^11.8.1",
+    "@emotion/styled": "^11.8.1",
+    "@ezcater/snitches": "^0.2.7",
```
[the remark this answers is not in the document]
Comment (noranda) [reply package.json:120]: This looks to be in error. I will fix!

File: src/components/EzButton/EzButton.tsx (hunk 2 above)
Comment (noranda) [reply src/components/EzButton/EzButton.tsx:267]: Will do!

**Review by pgrippi — APPROVED** [review]

**Review by danidewitt — APPROVED** [review]

**Pull request merged**